<!DOCTYPE html>
<html>
<head>
<meta charset="UTF-8">
<title>비동기작업</title>
</head>
<body>
<h1>Promise - 비동기 작업을 실행하는 중에 오류가 발생했음을 통지하기</h1>
<script>
const promise = new Promise((resolve, reject) => {
  var xhr = new XMLHttpRequest();
  xhr.onreadystatechange = function() {
    if (xhr.readyState == 4) {
      console.log(xhr.responseText);
      reject(); // 작업 수행을 정상적으로 완료하지 못했음을 통지한다.
    }
  };
  xhr.open("GET", "exam01", true); 
  xhr.send();
});

console.log("1111");

// promise 객체가 작업 거절 통지를 받았다 하더라도 
// 그 거절에 대한 처리를 수행할 함수가 등록되어 있지 않다면 
// 예외가 발생한다.


</script>
</body>
</html>